### ISSUE TITLE
It "should Set-Content to testdrive:\dynamicfile.txt with dynamic parameters" (but doesn't)

### ISSUE THREAD

**Issue by brycem** [PowerShell/PowerShell#891]
The following new DRT tests ported from psl-monad DRT TTests are passing on Windows, but failing on Linux\OSX:

```
        It "should Set-Content to testdrive:\dynamicfile.txt with dynamic parameters" -Pending:($IsLinux -Or $IsOSX) {
            $result=set-content -path testdrive:\dynamicfile.txt -value "ExpectedContent" -passthru
            $result| Should BeExactly "ExpectedContent"
        }
        It "should return expected string from testdrive:\dynamicfile.txt" {
            $result = get-content -path testdrive:\dynamicfile.txt
            $result | Should BeExactly "ExpectedContent"
        }
```

Here's an example of the stack output (note the substring '/tmp/dce3b97c-db38-41a3-bf1b-a
43372cbe454534' appears three times in the invalid path):

```
432add-content : Could not find a part of the path '/tmp/dce3b97c-db38-41a3-bf1b-a
43372cbe454534/tmp/dce3b97c-db38-41a3-bf1b-a72cbe454534/tmp/dce3b97c-db38-41a3-bf1
434b-a72cbe454534/dynamicfile.txt'.
435At /Users/travis/build/PowerShell/PowerShell/test/powershell/Add-Content.Tests.
436ps1:15 char:21
437+ ...     $result=add-content -path testdrive:\dynamicfile.txt -value "Expe ...
438+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
439    + CategoryInfo          : ObjectNotFound:
444
445    [-] should Add-Content to testdrive:\dynamicfile.txt with dynamic parameters 325ms
446      Expected exactly: {ExpectedContent}
447      But was: {}
448      at line: 16 in /Users/travis/build/PowerShell/PowerShell/test/powershell/Add-Content.Tests.ps1
449      16:             $result| Should BeExactly "ExpectedContent"
432add-content : Could not find a part of the path '/tmp/dce3b97c-db38-41a3-bf1b-a
43372cbe454534/tmp/dce3b97c-db38-41a3-bf1b-a72cbe454534/tmp/dce3b97c-db38-41a3-bf1
434b-a72cbe454534/dynamicfile.txt'.
435At /Users/travis/build/PowerShell/PowerShell/test/powershell/Add-Content.Tests.
436ps1:15 char:21
437+ ...     $result=add-content -path testdrive:\dynamicfile.txt -value "Expe ...
438+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
439    + CategoryInfo          : ObjectNotFound:
440(/tmp/dce3b97c-d...dynamicfile.txt:String) [Add-Content],
441DirectoryNotFoundException
442    + FullyQualifiedErrorId : GetContentWriterDirectoryNotFoundError,Microsoft.
443PowerShell.Commands.AddContentCommand
444
445    [-] should Add-Content to testdrive:\dynamicfile.txt with dynamic parameters 325ms
446      Expected exactly: {ExpectedContent}
447      But was: {}
448      at line: 16 in /Users/travis/build/PowerShell/PowerShell/test/powershell/Add-Content.Tests.ps1
449      16:             $result| Should BeExactly "ExpectedContent"
440(/tmp/dce3b97c-d...dynamicfile.txt:String) [Add-Content],
441DirectoryNotFoundException
442    + FullyQualifiedErrorId : GetContentWriterDirectoryNotFoundError,Microsoft.
443PowerShell.Commands.AddContentCommand
444
445    [-] should Add-Content to testdrive:\dynamicfile.txt with dynamic parameters 325ms
446      Expected exactly: {ExpectedContent}
447      But was: {}
448      at line: 16 in /Users/travis/build/PowerShell/PowerShell/test/powershell/Add-Content.Tests.ps1
449      16:             $result| Should BeExactly "ExpectedContent"
```

Note:  Same issue happens in add-content, so, I'm adding this to affected tests in add-content.tests.ps1 and set-content.tests.ps1:
`-Pending:($IsLinux -Or $IsOSX) {#https://github.com/PowerShell/PowerShell/issues/891`

**Comment by JamesWTruher** [comment]
This is really a provider issue, not a test problem.

**Comment by JamesWTruher** [comment]
this bug was fixed in October by Charu (commit ee2feaa)
